src/config/authd-config.c

```diff
@@ -298,6 +298,10 @@ int w_read_force_config(XML_NODE node, authd_config_t *config) {
             config->force_options.key_mismatch = b;
         }
         // disconnected_time
+        else if (!strcmp(node[i]->element, "force_time")){
```

Comment:
This condition cannot be included at `w_read_force_config` because this function reads the content of the block `<force>` but the old `<force_time>` is outside this block.

In addition, we need to set `disconnected_time` according to the value of `force_time`. When `force_time` is 0, we set `disconnected_time_enabled` to false. Otherwise, the value of `force_time` is stored at `disconnected_time`.

We need to do the same as `legacy_force_insert` does. Once read the whole `<authd>` block, we have to overwrite the value only if the old tag `force_time` was found, and also the new block `<force>` was not read. That means the configuration comes from an older version and has not been updated.